src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -359,7 +368,50 @@ def window(
         start_pos = self.to_internal_index(start)
         end_pos = self.to_internal_index(end)
 
-        return self._wrapped_buffer_window(self._buffer, start_pos, end_pos, force_copy)
+        window = self._wrapped_buffer_window(
+            self._buffer, start_pos, end_pos, force_copy
+        )
+
+        if fill_value is not None:
+            # We don't want to modify the original buffer
+            assert force_copy
```

Comment:
I think this assertion might be redundant as it fill_value and force_copy are checked at the beginning of the function

Reply:
Right, I had it to stress this here again. But probably not necessary. Removed in the update.